### PR TITLE
[ATO-663] Port solution to compress custom action request to main

### PR DESCRIPTION
**Pull request by radovanZRasa**
**Proposed changes**:
Add capability to compress body of an HTTP request for action server

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Review by ancalita — CHANGES_REQUESTED** [review]
I've left my review on `3.2.x` PR, the comments are applicable here too.

**Comment by radovanZRasa** [comment]
> I've left my review on `3.2.x` PR, the comments are applicable here too.

All comments on [Compress body of HTTP request (3.2.x)](https://github.com/RasaHQ/rasa/pull/11995) have been addressed.
All commits have been ported.

File: docs/docs/action-server/actions.mdx [hunk 1]
```diff
@@ -19,6 +19,18 @@ using a default location if the user doesn't specify one.
...
+:::info Compressed body in HTTP requests
+Rasa has capability to send compressed body in HTTP requests for custom actions.
+By default, this option is off to keep backward compatibility with older versions of
+custom action servers which did not receive compressed body in HTTP request for custom actions.
+To enable this option set environment variable COMPRESS_ACTION_SERVER_REQUEST to True.
+
+Rasa SDK 3.2.2, 3.3.1, 3.4.1 and onwards,
+support both compressed and non-compressed body in HTTP request for
+running custom action server. There is no additional setup required.
+
```
Comment (ancalita) [inline docs/docs/action-server/actions.mdx:32]: Missing closing info note
```suggestion
:::
```

File: docs/docs/action-server/actions.mdx [hunk 2]
```diff
@@ -19,6 +19,18 @@ using a default location if the user doesn't specify one.
...
 The message returned will differ according to the channel they are using. 
 
+
+:::info Compressed body in HTTP requests
+Rasa has capability to send compressed body in HTTP requests for custom actions.
+By default, this option is off to keep backward compatibility with older versions of
+custom action servers which did not receive compressed body in HTTP request for custom actions.
+To enable this option set environment variable COMPRESS_ACTION_SERVER_REQUEST to True.
+
+Rasa SDK 3.2.2, 3.3.1, 3.4.1 and onwards,
```
Comment (ancalita) [inline docs/docs/action-server/actions.mdx:29]:
```suggestion
Rasa has the capability to compress the HTTP request body for custom actions.
By default, this option is off to keep backward compatibility with older versions of
custom action servers which did not receive the compressed body in the HTTP request for custom actions.
To enable this option, set environment variable COMPRESS_ACTION_SERVER_REQUEST to `True`.

Rasa SDK versions `3.2.2`, `3.3.1`, `3.4.1` and upwards,
```

File: docs/docs/action-server/actions.mdx [hunk 3]
```diff
@@ -19,6 +19,18 @@ using a default location if the user doesn't specify one.
...
+support both compressed and non-compressed body in HTTP request for
```
Comment (ancalita) [inline docs/docs/action-server/actions.mdx:30]:
```suggestion
support both compressed and non-compressed body in the HTTP request for
```

**Review by ancalita — APPROVED** [review]
Left some docs suggestions.

**Pull request merged**